sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/spanner/SpannerIOReadTest.java

```diff
@@ -373,19 +379,12 @@ public void testReadMetrics() throws Exception {
             any(ReadQueryUpdateTransactionOption.class)))
         .thenReturn(Arrays.asList(fakePartition, fakePartition, fakePartition));
     when(mockBatchTx.execute(any(Partition.class)))
-        .thenThrow(
```

Comment:
Should we also keep the exception in this test and catch it ?

Reply:
It gets caught below - the try/catch block around pipeline.run().

The previous behaviour of SpannerIO.Read was to fail the pipeline completely when Read threw an exception (in fact runners like Dataflow would retry the bundle a few times first). This PR restores that original behaviour (whether or not it is the ideal behaviour is another question!)

Reply:
Not in this test (testReadMetrics) ?

Reply:
BTW previous behavior is correct. We should fail the work item for non-retriable errors and such failures will usually be retried few times by the runner.

Reply:
Added and renamed tests: 

testQueryMetricsFail()
testQueryMetricsSucceed()
testReadMetricsFail()
testReadMetricsSucceed()